buildSrc/src/test/java/io/micronaut/guides/core/ResourceMacroSubstitutionTest.java

```diff
@@ -0,0 +1,78 @@
+package io.micronaut.guides.core;
+
+import io.micronaut.starter.api.TestFramework;
+import io.micronaut.starter.options.BuildTool;
+import io.micronaut.starter.options.Language;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@MicronautTest(startApplication = false)
+public class ResourceMacroSubstitutionTest {
+
+    @Inject
+    ResourceMacroSubstitution resourceMacroSubstitution;
+
+    @Test
+    void testSubstitute() {
+        String str = "resource:../../../ttfr.sh[]";
+        String resJava = resourceMacroSubstitution.substitute(str, "executable-jar", new GuidesOption(BuildTool.GRADLE, Language.JAVA, TestFramework.JUNIT));
+        String expectedJava = """
+                [source,]
+                .ttfr.sh
+                ----
+                include::{sourceDir}/executable-jar/executable-jar-gradle-java/src/main/resources/../../../ttfr.sh[]
+                ----""";
+        assertEquals(expectedJava, resJava);
+    }
+
+    @Test
+    void testSubstituteWithTags() {
+        String str = "resource:application.yml[tag=endpoints]";
+        String resJava = resourceMacroSubstitution.substitute(str, "adding-commit-info", new GuidesOption(BuildTool.GRADLE, Language.JAVA, TestFramework.JUNIT));
+        String expectedJava = """
+                [source,yaml]
+                .src/main/resources/application.yml
+                ----
+                include::{sourceDir}/adding-commit-info/adding-commit-info-gradle-java/src/main/resources/application.yml[tag=endpoints]
+                
```

Comment:
please remove this extra blank line

Reply:
You confirm you want it removed? It is not a bug of my code but the current implementation of AsciidocGenerator adds that line. Here for example _adding-commit-info-maven-groovy.adoc_
<img width="766" alt="Screenshot 2024-10-25 at 11 35 25" src="https://github.com/user-attachments/assets/58798416-4600-4b1d-86b6-b07d01e1f782">

Reply:
Yes, I want it removed.